type-generator/src/main/java/io/mesosphere/types/dtr/compiler/impl/StringTemplateEngine.java

```diff
@@ -0,0 +1,777 @@
+package io.mesosphere.types.dtr.compiler.impl;
+
+import io.mesosphere.types.dtr.compiler.internal.*;
+import io.mesosphere.types.dtr.compiler.internal.arguments.PrecalcValueArguments;
+import io.mesosphere.types.dtr.compiler.internal.arguments.TypeArguments;
+import io.mesosphere.types.dtr.compiler.internal.arguments.ValueArguments;
+import io.mesosphere.types.dtr.models.internal.Type;
+import io.mesosphere.types.dtr.models.internal.TypeScope;
+import io.mesosphere.types.dtr.models.internal.Value;
+import io.mesosphere.types.dtr.models.internal.types.*;
+import io.mesosphere.types.dtr.models.internal.values.AnyValue;
+import io.mesosphere.types.dtr.models.internal.values.DateTimeValue;
+import io.mesosphere.types.dtr.repository.internal.RepositoryView;
+import org.antlr.runtime.Token;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringEscapeUtils;
+import org.ini4j.Ini;
+import org.stringtemplate.v4.ST;
+import org.stringtemplate.v4.STGroup;
+import org.stringtemplate.v4.STGroupString;
+import org.stringtemplate.v4.StringRenderer;
+import org.stringtemplate.v4.misc.Misc;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * A StringTemplateEngine engine is using the StringTemplateEngine template engine to
+ * compile a user expression into a buffer.
+ */
+public class StringTemplateEngine implements CompilerEngine {
+
+    /**
+     * The zone ID to use when parsing/writing dates
+     */
+    private ZoneId dateZone;
+
+    /**
+     * Mapping of scalar types
+     */
+    private Map<String, String> scalarTypes;
```

Comment:
What does that mean?

Reply:
Haha, true. Each scalar type (well. `Value` instances to be precise) has an internal name, that needs to be mapped to the language-specific type name. For instance:

```ini
any=Object
binary=byte[]
boolean=Boolean
datetime=java.time.OffsetDateTime
date_only=java.time.OffsetDateTime
time_only=java.time.OffsetDateTime
datetime_only=java.time.OffsetDateTime
integer=Int
number=Double
string=String
void=None
```

Reply:
I think this should be done in the string template itself. It does support maps, right?